lib/execution_engine2/db/MongoUtil.py

```diff
@@ -263,6 +267,68 @@ def check_if_already_finished(job_status):
             return True
         return False
 
+    def update_jobs_to_queued(
+        self, job_id_pairs: List[JobIdPair], scheduler_type: str = "condor"
+    ) -> None:
+        f"""
+        * Adds scheduler id to list of jobs
+        * Updates a list of {Status.created.value} jobs to queued. Does not work on jobs that already have gone through any other
+        status transition. If the record is not in the {Status.created.value} status, nothing will happen
+        :param job_id_pairs: A list of pairs of Job Ids and Scheduler Ids
+        :param scheduler_type: The scheduler this job was queued in, default condor
+        """
+
+        bulk_update_scheduler_jobs = []
+        bulk_update_created_to_queued = []
+        queue_time_now = datetime.utcnow().timestamp()
+        for job_id_pair in job_id_pairs:
+            if job_id_pair.job_id is None:
+                raise ValueError(
+                    f"Provided a bad job_id_pair, missing job_id for {job_id_pair.scheduler_id}"
+                )
+            elif job_id_pair.scheduler_id is None:
+                raise ValueError(
+                    f"Provided a bad job_id_pair, missing scheduler_id for {job_id_pair.job_id}"
+                )
+
+            bulk_update_scheduler_jobs.append(
+                UpdateOne(
+                    {
+                        "_id": ObjectId(job_id_pair.job_id),
+                    },
+                    {
+                        "$set": {
+                            "queued": queue_time_now,
```

Comment:
Right, but `test_run_job_batch` calls `_get_mongo_job` and `assert_close_to_now(job.pop("updated"))` but I can add something similar in here. Done

Reply:
Yeah, I'm always going to ask for unit tests when it's reasonably possible to do since they're usually a lot easier to understand and debug than integration tests.

Anyway :+1: